Project/DAQ_System/Program/Src/app.cpp

```diff
@@ -153,5 +172,49 @@ void cppMain() {
 	}
 }
 
+osThreadId_t dataLoggingTaskHandle;
+const osThreadAttr_t dataLoggingTask_attributes = {
+  .name = "dataLoggingTask",
+  .stack_size = 128 * 17,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+
+void RtosInit() {
+	dataLoggingTaskHandle = osThreadNew(DataLoggingThread, NULL, &dataLoggingTask_attributes);
+}
```

Comment:
kinda confused here. So is this where we put threads that we want to manage? It looks like you are creating and initializing the thread above this RtosInit(). So in theory we would first create a thread, initialize the thread(name,stack size, priority,etc), Place it in a queue with RtosInit() as well as create the actual instructions/ task in that thread? Then form here, you can do osKernelStart() to start scheduling and managing the threads that were initialized with Rtos from RtosInit()?

Reply:
> So is this where we put threads that we want to manage?

Maybe. For now, this will do. It would definitely be easiest to integrate the threads in our main file where all the components are integrated. 

> It looks like you are creating and initializing the thread above this RtosInit()

That's right! `osThreadNew()` expects a function pointer (here, I am passing the address of where the `DataLoggingThread()` function is defined in flash memory), a pointer to arguments (here, `NULL`), and the thread's attributes (defined in line 176).

> So in theory we would first create a thread, initialize the thread(name,stack size, priority,etc), Place it in a queue with RtosInit() as well as create the actual instructions/ task in that thread?

I picture the boot-up process to be like this:

- Delegate threads for each firmware component
- Initialize and run the threads
- All sensors would have access a struct holding the data payload, updating its values appropriately
- Periodically, a timestamp will be created and will be associated with a copy of the data payload struct, inserted into a queue
- The Data Logger will check for whether there is a new batch of data samples in the queue. If so, log it.